server/src/main/java/org/opensearch/transport/TransportService.java

```diff
@@ -931,7 +936,7 @@ private <T extends TransportResponse> void sendRequestInternal(
                 // thread on a best effort basis though.
                 final SendRequestTransportException sendRequestException = new SendRequestTransportException(node, action, e);
                 final String executor = lifecycle.stoppedOrClosed() ? ThreadPool.Names.SAME : ThreadPool.Names.GENERIC;
-                threadPool.executor(executor).execute(new AbstractRunnable() {
+                threadPool.executor(executor).execute(MyShiroModule.getSubjectOrInternal().associateWith(new AbstractRunnable() {
```

Comment:
qq, is this for logging which user was associated with this action or is it something else entirely?

Reply:
When using `execute(...)` a new thread is created that runs the 'Runnable' code, Shiro needs to have its state sent into the new thread and this `associateWith` call ensures this takes place